### PR TITLE
major version upgrade

### PR DESCRIPTION
**Pull request by mechanicMo**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,22 +28,36 @@ const saveAs = (uri, filename) => {
...
         useCORS: true,
-        ...options
+        ...html2CanvasOptions
     }).then(canvas => {
         if (type === fileType.PDF) {
-            const pdf = canvas.width > canvas.height
-                ? new JsPDF('l', 'mm', [canvas.width, canvas.height])
-                : new JsPDF('p', 'mm', [canvas.height, canvas.width]);
+            const pdf = getPDF(type, pdfOptions)
             pdf.addImage(canvas.toDataURL(fileType.PNG, 1.0), 'PNG', 0, 0);
```
Comment (salman-monetate) [inline index.js:61]: x, y from pdf options will be used here.

File: README.md [hunk 2]
```diff
@@ -27,13 +46,13 @@ export default class MyComponent extends React.Component {
     return (
       <React.Fragment>
         <ComponentToPrint ref={this.componentRef} />
-        <button onClick={() => exportComponentAsJPEG(this.componentRef)}>
+        <button onClick={() => exportComponentAsJPEG({node: this.componentRef})}>
```
Comment (salman-monetate) [inline README.md:49]: `node`  should still be passed directly otherwise it will break existing usage.

File: index.js (hunk 1 above)
Comment (mechanicMo) [reply index.js:63]: i addressed this in the latest commit

File: README.md (hunk 2 above)
Comment (mechanicMo) [reply README.md:49]: i addressed this in the latest commit

**Comment by mechanicMo** [comment]
also, i removed type as per your suggestion

File: index.js [hunk 3]
```diff
@@ -28,58 +29,107 @@ const saveAs = (uri, filename) => {
...
  * @param  {string} fileName
- * @param  {string} backgroundColor
  * @param  {string} type
- * @param  {object} options=null
+ * @param  {object} html2CanvasOptions={}
  */
-const exportComponent = (node, fileName, backgroundColor, type, options) => {
+
+const getPDF = (canvas, {x, y, orientation}) => {
+    const width = x || canvas.width
```
Comment (salman-monetate) [inline index.js:37]: This x, y I believe is useful in the alignment when we call `pdf.addImage(canvas.toDataURL(fileType.PNG, 1.0), 'PNG', x, y);`

https://artskydj.github.io/jsPDF/docs/module-addImage.html

Comment (mechanicMo) [reply index.js]: i read through the docs and made additional changes. let me know what you think

**Comment by salman-monetate** [comment]
You can build and push the code with version update in the package.

**Review by salman-monetate — APPROVED** [review]

**Comment by mechanicMo** [comment]
> You can build and push the code with version update in the package. Let me know if you want me to do it.

When i try to build it, it says that the asset sizes are over the recommended limit. 

Does it work if you change line 61 to:

const element = node.current

That way you can remove reactDom all together

**Comment by salman-monetate** [comment]
> > You can build and push the code with version update in the package. Let me know if you want me to do it.
> 
> When i try to build it, it says that the asset sizes are over the recommended limit.
> 
> Does it work if you change line 61 to:
> 
> const element = node.current
> 
> That way you can remove reactDom all together

I haven't tried that. There is bundle analyzer added to the package if you run `build:analyze` it will show that the majority is `jspdf` and `html2Canvas`.

**Comment by mechanicMo** [comment]
> > > You can build and push the code with version update in the package. Let me know if you want me to do it.
> > 
> > 
> > When i try to build it, it says that the asset sizes are over the recommended limit.
> > Does it work if you change line 61 to:
> > const element = node.current
> > That way you can remove reactDom all together
> 
> I haven't tried that. There is bundle analyzer added to the package if you run `build:analyze` it will show that the majority is `jspdf` and `html2Canvas`.

okay, i have pushed the build and package # update

**Pull request merged**